src/css/parse.js

```diff
@@ -828,30 +903,36 @@ function getBrackets() {
  * @return {Number} Length of the class selector
  */
 function checkClass(i) {
-  var l;
+  const start = i;
+  let l;
 
   if (i >= tokensLength) return 0;
 
-  if (tokens[i].class_l) return tokens[i].class_l;
+  if (tokens[i++].type !== TokenType.FullStop) return 0;
 
-  if (tokens[i++].type === TokenType.FullStop && (l = checkIdent(i))) {
-    tokens[i].class_l = l + 1;
-    return l + 1;
-  }
+  if (l = checkIdent(i)) i += l;
+  else return 0;
 
-  return 0;
+  tokens[start].classEnd = i;
+
+  return i - start;
 }
 
 /**
  * Get node with a class selector
  * @return {Node}
  */
 function getClass() {
-  let type = NodeType.ClassType;
-  let token = tokens[pos++];
-  let line = token.ln;
-  let column = token.col;
-  let content = [getIdent()];
+  const type = NodeType.ClassType;
+  const token = tokens[pos];
+  const line = token.ln;
+  const column = token.col;
+  let content = [];
+
+  // Skip `.`
+  pos++;
+
+  content = content.concat(getIdent());
```

Comment:
I'll clean this up but `concat` can merge both [arrays and objects](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Array/concat). With that in mind, do we still want to wrap objects in arrays like the following? 
```js
const content = [].concat(
    getProperty(),
    getSC(),
    [getPropertyDelim()],
    getSC(),
    [getValue()]
  );
```
The code below is valid;
```js
const content = [].concat(
    getProperty(),
    getSC(),
    getPropertyDelim(),
    getSC(),
    getValue()
  );
```
Up to you :+1: